sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,313 @@
+# Authors: Nicolas Goix <nicolas.goix@telecom-paristech.fr>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.stats import scoreatpercentile
+
+from .base import NeighborsBase
+from .base import KNeighborsMixin
+from .base import UnsupervisedMixin
+
+from ..metrics import pairwise_distances
+
+from ..utils.validation import check_is_fitted
+from ..utils import check_array
+
+__all__ = ["LocalOutlierFactor"]
+
+
+class LocalOutlierFactor(NeighborsBase, KNeighborsMixin, UnsupervisedMixin):
+    """Unsupervised Outlier Detection using Local Outlier Factor (LOF)
+
+    The anomaly score of each sample is called Local Outlier Factor.
+    It measures the local deviation of density of a given sample with
+    respect to its neighbors.
+    It is local in that the degree depends on how isolated the object is
+    with respect to the surrounding neighborhood.
+    More precisely, locality is given by k-nearest neighbors, whose distance
+    is used to estimate the local density.
+    By comparing the local density of a sample to the local densities of
+    its neighbors, one can identify samples that have a substantially lower
+    density than their neighbors. These are considered as outliers.
+
```

Comment:
I'd add the reference of the LOF paper.

Also LOF of inliers should be close to 1 according to LOF paper. So here the decision function values should be close to -1 for inliers. Could be a good thing to mention this point.